### DISCUSSION TITLE
Add ability to drag beatmap links into the game

### DISCUSSION THREAD

**Discussion by respektive** [ppy/osu#17214] [Ideas]
I use this quite a lot on stable, when i drag links from discord bots and such into the game to immediately jump to the map or download via direct. It's faster like that, than opening it in browser and clicking the direct link and then having to close the tab again.

I believe it was mentioned somewhere before, that this was planned to be added, but I couldn't find any discussion/issue.

**Comment by peppy** [comment]
We already have drag drop support for files, but URL dropping is a different implementation path which we don't have yet.

**Reply by frenzibyte** [reply]
Adding support for such on macOS seems to be pretty simple using the native interop logic we have in place (`registerForDraggedTypes:` and listening via `draggingEntered`). Unsure about Windows.

**Reply by bdach** [reply]
i would try [`SDL_DROPTEXT`](https://wiki.libsdl.org/SDL_DropEvent) first.

**Reply by frenzibyte** [reply]
Forgot to mention, `SDL_DROPTEXT` was my immediate thought, but that turned out not to work at all on macOS, perhaps Windows only?

**Reply by frenzibyte** [reply]
In fact, upon a closer look, `SDL_SendDropText` is only ever called on x11 environments:

<img width="601" alt="CleanShot 2022-04-07 at 16 46 17@2x" src="https://user-images.githubusercontent.com/22781491/162213825-9fcba2d0-382e-4dbe-9151-427a58687e8a.png">

So neither Windows nor macOS support receiving `SDL_DROPTEXT` events at all.
